zenpacklib.py

```diff
@@ -806,14 +819,39 @@ def get_facets(self, seen=None):
                 # This is really a single object.
                 relobjs = [relobjs]
 
+            relpath = "/".join(path + [relname])
+
+            # Always include directly-related objects.
             for obj in relobjs:
                 if obj in seen:
                     continue
 
                 yield obj
                 seen.add(obj)
-                for facet in obj.get_facets(seen=seen):
+
+            # If 'all' mode, just include indirectly-related objects as well, in
+            # an unfiltered manner.
+            if recurse_all:
+                for facet in obj.get_facets(root=root, seen=seen, path=path + [relname], recurse_all=True):
                     yield facet
+                return
+
+            # Otherwise, look at extra_path defined path pattern streams
+            for stream in streams:
+                recurse = any([pattern.match(relpath) for pattern in stream])
+
+                LOG.debug("[%s] matching %s against %s: %s" % (root.meta_type, relpath, [x.pattern for x in stream], recurse))
```

Comment:
This seems like useful logging for the ZenPack author, but considering how often our users enable debug logging I think this will be too noisy. Maybe do `LOG.log(9, "msg", ...)` instead? That way people would have to do -v9 or lower to see this level of logging? We've done this before with the WindowsMonitor ZenPack to enable "developer" level logging.